Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -152,6 +155,19 @@
             m_remoteLayerTreeHost->detachRootLayer();
     }
 
+    for (auto& [layerID, properties] : layerTreeTransaction.changedLayerProperties()) {
```

Comment:
This seems like an odd place to trigger geometry changes; let's at least factor into `RemoteLayerTreeDrawingAreaProxy::updateSubframeGeometry()`.

Layer bounds is not the right way to get the size of a frame; this data needs to come from the frame tree.

Reply:
Could you point me more clearly to where you think this information should come from?  We need to get the size post-layout and this seemed to be the place where it ends up in the hosting layer size, which is what we want to put the iframe contents into.

Reply:
This depends on what DrawingAreaProxy–DrawingArea relationship we end up with for subframes. But you should consider the layer tree as "presentation-only"; we only ever push to it, never read back geometry from it.